### ISSUE TITLE
[SERVER ISSUE] Server crashes due to OutOfMemoryException, possible memory leak? [7.1.2]

### ISSUE THREAD

**Issue by Ekkelus** [StrangeLoopGames/EcoIssues#4573]
Modded server, made changes to EcoSim.eco and installed CTK. This server breaking issue started when we updated to the latest patch. Just before one of the crashes there were only 2 players on and one of them was mining.

World is 140 x 140.

**Server Specs:**
Intel Xeon CPU E3-1265L V2 @ 2.50Ghz
16 GB RAM

I noticed that when there were only 3 people on RAM usage was at 60%, usually not like that.

` --BEGIN DUMP--
Dump Time
02/15/2018 06:39:09

Exception
System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at Eco.Simulation.WorldLayers.Pushers.PlantSpawner.Apply(WorldRange area, Single[] layerValues, WorldLayerNeighborInfo[] neighborValues)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.<>c__DisplayClass141_0.<PushLayersAtMyGranularity>b__0(Vector2i pos, WorldRange area)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.ForEachCell(Action`2 action)
   at Eco.Simulation.WorldLayers.NewWorldLayerSync.Push()
   at Eco.Simulation.WorldLayers.WorldLayerSync.<>c.<Tick>b__11_5()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerSync.Tick()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerManager.Tick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.TryTick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)

--END DUMP-- `
` --BEGIN DUMP--
Dump Time
02/15/2018 08:48:06

Exception
System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at Eco.Simulation.WorldLayers.Pushers.PlantSpawner.Apply(WorldRange area, Single[] layerValues, WorldLayerNeighborInfo[] neighborValues)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.<>c__DisplayClass141_0.<PushLayersAtMyGranularity>b__0(Vector2i pos, WorldRange area)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.ForEachCell(Action`2 action)
   at Eco.Simulation.WorldLayers.NewWorldLayerSync.Push()
   at Eco.Simulation.WorldLayers.WorldLayerSync.<>c.<Tick>b__11_5()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerSync.Tick()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerManager.Tick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.TryTick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)

--END DUMP--
`
` --BEGIN DUMP--
Dump Time
02/15/2018 09:05:28

Exception
System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at Eco.Simulation.WorldLayers.Pushers.PlantSpawner.Apply(WorldRange area, Single[] layerValues, WorldLayerNeighborInfo[] neighborValues)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.<>c__DisplayClass141_0.<PushLayersAtMyGranularity>b__0(Vector2i pos, WorldRange area)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.ForEachCell(Action`2 action)
   at Eco.Simulation.WorldLayers.NewWorldLayerSync.Push()
   at Eco.Simulation.WorldLayers.WorldLayerSync.<>c.<Tick>b__11_5()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerSync.Tick()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerManager.Tick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.TryTick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)

--END DUMP--
`

**Comment by Ekkelus** [comment]
So the server was working fine yesterday but all of a sudden today it started crashing once every 20 minutes or so with the following dump:


--BEGIN DUMP--
Dump Time
02/16/2018 15:54:11

Exception
System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.Collections.Generic.List1.set_Capacity(Int32 value)
   at System.Collections.Generic.List1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List1.Add(T item)
   at Eco.Simulation.WorldLayers.Pushers.PlantSpawner.Apply(WorldRange area, Single[] layerValues, WorldLayerNeighborInfo[] neighborValues)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.<>c__DisplayClass141_0.<PushLayersAtMyGranularity>b__0(Vector2i pos, WorldRange area)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.ForEachCell(Action2 action)
   at Eco.Simulation.WorldLayers.NewWorldLayerSync.Push()
   at Eco.Simulation.WorldLayers.WorldLayerSync.<>c.<Tick>b__11_5()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerSync.Tick()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerManager.Tick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.TryTick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)

--END DUMP--

**Comment by Ekkelus** [comment]
World file if that helps:
https://ufile.io/p08zx

**Comment by Zhaikadom** [comment]
I'm experiencing the same OOM Exception:
--BEGIN DUMP--
Dump Time
03/14/2018 14:56:08

Exception
System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at Eco.Simulation.WorldLayers.Pushers.PlantSpawner.Apply(WorldArea area, Single[] layerValues, WorldLayerNeighborInfo[] neighborValues)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.<>c__DisplayClass141_0.<PushLayersAtMyGranularity>b__0(Vector2i pos, WorldArea area)
   at Eco.Simulation.WorldLayers.Layers.WorldLayer.ForEachCell(Action`2 action)
   at Eco.Simulation.WorldLayers.NewWorldLayerSync.Push()
   at Eco.Simulation.WorldLayers.WorldLayerSync.<>c.<Tick>b__11_5()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerSync.Tick()
   at Eco.Core.Plugins.TickTimeUtil.TimeSubprocess(Action func)
   at Eco.Simulation.WorldLayers.WorldLayerManager.Tick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.TryTick()
   at Eco.Simulation.WorldLayers.WorldLayerManager.Run()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart(Object obj)

--END DUMP--

The issue originally appeared on a hosted environment, I also copied required files to a local server, and the same issue occurs. The server usually crashes after around 10 minutes.

The server is modded, I've included the config, mods and storage directories in the archive (along with one of the dump files):
https://drive.google.com/open?id=1DtZnLdlyQx74Jh59a6_e5vZQS4XYOn53
